independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java

```diff
@@ -162,11 +163,25 @@ Collection<Resource> generate(String name, BeanDeployment beanDeployment, Map<Be
                     getComponents.load(entry.getKey().toString()), nonbindingMembers);
         }
 
+        ResultHandle contextInstances;
+        if (scopeToContextInstances.isEmpty()) {
+            contextInstances = getComponents.invokeStaticMethod(MethodDescriptors.COLLECTIONS_EMPTY_MAP);
+        } else {
+            contextInstances = getComponents.newInstance(MethodDescriptor.ofConstructor(HashMap.class));
+            for (Entry<DotName, String> e : scopeToContextInstances.entrySet()) {
+                ResultHandle scope = getComponents.loadClass(e.getKey().toString());
+                ResultHandle supplier = getComponents.invokeStaticMethod(
+                        MethodDescriptor.ofMethod(Components.class, "newContextInstancesSupplier", Supplier.class, Class.class),
+                        getComponents.loadClass(e.getValue()));
```

Comment:
We could generate implementations of `Supplier<ContextInstances>` here (which would basically just call the zero-param ctor of `e.getValue()` in `Supplier.get`, so pretty simple Gizmo stuff) instead of the ugly reflective `newContextInstancesSupplier()`. Not a big deal I guess.

Reply:
Good idea!